File: Signiture-Backend/delivery/api/src/main/java/pcy/study/api/common/api/ApiResult.java
package pcy.study.api.common.api;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;

@Builder
public record ApiResult(
        @Schema(description = "코드")
        Integer code,
        @Schema(description = "메시지")
        String message
) {

    public static ApiResult ok() {
        return create(ResponseCode.OK);
    }

    public static ApiResult error(ApiCode apiCode) {
        return create(apiCode);
    }

    private static ApiResult create(ApiCode apiCode) {
        return ApiResult.builder()
                .code(apiCode.getCode())
                .message(apiCode.getMessage())
                .build();
    }
}

File: Signiture-Backend/kotlin/delivery/common/src/main/kotlin/pcy/study/common/api/ApiResult.kt
package pcy.study.common.api

import pcy.study.common.api.code.ResponseCode

data class ApiResult(
    val code: Int,
    val message: String? = null,
) {

    companion object {

        fun ok(): ApiResult {
            return create(ResponseCode.OK)
        }

        fun error(apiCode: ApiCode): ApiResult {
            return create(apiCode)
        }

        fun error(
            code: Int,
            message: String,
        ): ApiResult {
            return ApiResult(
                code = code,
                message = message
            )
        }

        private fun create(apiCode: ApiCode): ApiResult {
            return ApiResult(
                code = apiCode.getCode(),
                message = apiCode.getMessage()
            )
        }
    }
}